cloud_experts_tutorials/cloud-experts-deploying-application/cloud-experts-deploying-application-integrating-aws.adoc

```diff
@@ -0,0 +1,391 @@
+namespace: ${OSTOY_NAMESPACE}
+spec:
+name: ${OSTOY_NAMESPACE}-bucket
+EOF
+----
++
+** Download the link:yaml/s3-bucket.yaml[command], then run it.
++
+[IMPORTANT]
+====
+The OSToy application expects to find a bucket named `<namespace\>-bucket`. If you use anything other than the namespace of your OSToy project, this feature will not work. For example, if our project is "ostoy", the value for `name` must be `ostoy-bucket`.
+====
++
+. Confirm the bucket was created by running the following command:
++
+[source,terminal]
+----
+$ aws s3 ls | grep ${OSTOY_NAMESPACE}-bucket
+----
+
+== Redeploying the OSToy app with the new service account
+
+. Run your pod with the service account you created.
+. Patch the `ostoy-frontend` deployment by running the following command:
++
+[source,terminal]
+----
+$ oc patch deploy ostoy-frontend -n ${OSTOY_NAMESPACE} --type=merge --patch '{"spec": {"template": {"spec":{"serviceAccount":"ostoy-sa"}}}}'
+----
++
+. Specify the service account:
++
+[source,terminal]
+----
+$ spec:
+# Uncomment to use with ACK portion of the workshop
+# If you chose a different service account name please replace it.
+serviceAccount: ostoy-sa
+containers:
+- name: ostoy-frontend
+image: quay.io/ostoylab/ostoy-frontend:1.6.0
+imagePullPolicy: IfNotPresent
+[...]
+----
++
+. Deploy the microservice by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f https://raw.githubusercontent.com/openshift-cs/rosaworkshop/master/rosa-workshop/ostoy/yaml/ostoy-microservice-deployment.yaml
+----
++
+. Wait for the pod to update.
+
+== Confirming the environment variables
+
+When AWS clients detect `AssumeRoleWithWebIdentity` security tokens to assume the IAM role to connect to the AWS APIs. See the link:https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html[AWS AssumeRoleWithWebIdentity] documentation for more details.
+
+* Use the following command to describe the pods and verify that the `AWS_WEB_IDENTITY_TOKEN_FILE` and `AWS_ROLE_ARN` environment variables exist for our application:
++
+[source,terminal]
+----
+$ oc describe pod ostoy-frontend -n ${OSTOY_NAMESPACE} | grep "^\s*AWS_"
+----
++
+.Example output
++
+[source,terminal]
+----
+AWS_ROLE_ARN:                 arn:aws:iam::000000000000:role/ostoy-sa
+AWS_WEB_IDENTITY_TOKEN_FILE:  /var/run/secrets/eks.amazonaws.com/serviceaccount/token
+----
+
+== Viewing the bucket contents through OSToy
+
+Use your app to view the contents of your S3 bucket.
+
+. Get the route for the newly deployed application by running the following command:
++
+[source,terminal]
+----
+$ oc get route ostoy-route -n ${OSTOY_NAMESPACE} -o jsonpath='{.spec.host}{"\n"}'
+----
++
+. Open a new browser tab and enter the route obtained in the previous step.
++
+[IMPORTANT]
+====
+Be sure to use `http://` and not `https://`.
+====
++
+. Click *ACK S3* in the left menu in OSToy.
+. Make sure that the page that lists the contents of the bucket is empty.
```

Comment:
I wouldn't say to make sure that the page is empty.  It is rather pointing out that since we have a new bucket it is empty.